roles/lib_openshift/src/lib/base.py

```diff
@@ -22,6 +22,29 @@ def __init__(self,
         self.verbose = verbose
         self.kubeconfig = Utils.create_tmpfile_copy(kubeconfig)
         self.all_namespaces = all_namespaces
+        self.oc_binary = self._locate_oc_binary()
+
+    @staticmethod
+    def _locate_oc_binary():
+        ''' Find and return oc binary file '''
+        # https://github.com/openshift/openshift-ansible/issues/3410
+        # oc can be in /usr/local/bin in some cases, but that may not
+        # be in $PATH due to ansible/sudo
+        paths = os.environ.get("PATH", os.defpath).split(os.pathsep) + ['/usr/local/bin', os.path.expanduser('~/bin')]
+        paths = list(set(paths))
```

Comment:
The way `which` and executable lookup works is directly dependent on and influenced by the order of the items in $PATH.